onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -2493,7 +2581,7 @@ CUDAExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph,
       continue;
 
     const auto& node = *p_node;
-    if (!node.GetExecutionProviderType().empty()) {
+    if (!node.GetExecutionProviderType().empty() && node.Domain() != kMSInternalNHWCDomain) {
```

Comment:
If I understand correctly, this logic is only safe as long as no other EP accepts `kMSInternalNHWCDomain` nodes. If there is another EP that starts accepting those nodes and that shows up ahead of the CUDA EP in the EP hierarchy list and if a node gets partitioned to that EP, when CUDA EP's `GetCapability()` is called, that node will not be skipped (it should get skipped here as the other EP has higher precedence than CUDA EP and that EP is able to run that node).